easybuild/easyblocks/t/tensorflow.py

```diff
@@ -701,13 +701,24 @@ def configure_step(self):
                 })
             else:
                 raise EasyBuildError("TensorFlow has a strict dependency on cuDNN if CUDA is enabled")
+
             if nccl_root:
                 nccl_version = get_software_version('NCCL')
                 # Ignore the PKG_REVISION identifier if it exists (i.e., report 2.4.6 for 2.4.6-1 or 2.4.6-2)
                 nccl_version = nccl_version.split('-')[0]
                 config_env_vars.update({
                     'NCCL_INSTALL_PATH': nccl_root,
                 })
+
+                # add absolute path to libnccl.so.2 directory provided by NCCL
+                # when LD_LIBRARY_PATH is filtered and LIBRARY_PATH is not
+                # filtered, e.g., in an environment such as EESSI
+                filtered_env_vars = build_option('filter_env_vars') or []
+                if 'LD_LIBRARY_PATH' in filtered_env_vars and 'LIBRARY_PATH' not in filtered_env_vars:
+                    system_libs_info_as_list = list(self.system_libs_info)
+                    system_libs_info_as_list[2].append(os.path.join(nccl_root, get_software_libdir('NCCL')))
+                    self.system_libs_info = tuple(system_libs_info_as_list)
```

Comment:
That part looks very fishy and I had to read the full code to verify this is correct. We should have `get_systems_libs` and hence `self.system_libs_info` a named tuple instead to make it easier to understand.

However from a semantic POV this is the wrong place to add NCCL: "System libs" in the context of tensorflow are dependencies that can be vendored in a way TF understands. I.e. https://github.com/tensorflow/tensorflow/blob/master/third_party/systemlibs/syslibs_configure.bzl#L11

I'd rather put this into the `build_step` where `action_env['LIBRARY_PATH']` is set. The easiest way would be to (conditionally)  append to `libpaths` right after `cpaths, libpaths = self.system_libs_info[1:]`

This is also easier to understand due to the comment there: "Make TF find our modules. LD_LIBRARY_PATH gets automatically added by configure.py"

Reply:
Sounds right to me.

Reply:
Made an attempt in https://github.com/easybuilders/easybuild-easyblocks/pull/3497/commits/f697d9749186d731ddb005c1c8220954919072d6

Not tested yet. Will let you know if it works or not.

Reply:
The changes in https://github.com/easybuilders/easybuild-easyblocks/commit/f697d9749186d731ddb005c1c8220954919072d6 worked.